libraries/plugins/elasticsearch/elasticsearch_plugin.cpp

```diff
@@ -0,0 +1,376 @@
+      flat_set<account_id_type> impacted;
+      vector<authority> other;
+      operation_get_required_authorities( op.op, impacted, impacted, other ); // fee_payer is added here
+
+      if( op.op.which() == operation::tag< account_create_operation >::value )
+         impacted.insert( op.result.get<object_id_type>() );
+      else
+         graphene::app::operation_get_impacted_accounts( op.op, impacted );
+
+      for( auto& a : other )
+         for( auto& item : a.account_auths )
+            impacted.insert( item.first );
+
+      for( auto& account_id : impacted )
+      {
+         add_elasticsearch( account_id, oho, b );
+      }
+   }
+}
+
+void elasticsearch_plugin_impl::add_elasticsearch( const account_id_type account_id, const optional <operation_history_object>& oho, const signed_block& b)
+{
+   graphene::chain::database& db = database();
+   const auto &stats_obj = account_id(db).statistics(db);
+
+   // add new entry
+   const auto &ath = db.create<account_transaction_history_object>([&](account_transaction_history_object &obj) {
+      obj.operation_id = oho->id;
+      obj.account = account_id;
+      obj.sequence = stats_obj.total_ops + 1;
+      obj.next = stats_obj.most_recent_op;
+   });
+
+   // keep stats growing as no op will be removed
+   db.modify(stats_obj, [&](account_statistics_object &obj) {
+      obj.most_recent_op = ath.id;
+      obj.total_ops = ath.sequence;
+   });
+
+   // operation_type
+   int op_type = -1;
+   if (!oho->id.is_null())
+      op_type = oho->op.which();
+
+   // operation history data
+   operation_history_struct os;
+   os.trx_in_block = oho->trx_in_block;
+   os.op_in_trx = oho->op_in_trx;
+   os.operation_result = fc::json::to_string(oho->result);
+   os.virtual_op = oho->virtual_op;
+   os.op = fc::json::to_string(oho->op);
+
+   // visitor data
+   visitor_struct vs;
+   if(_elasticsearch_visitor) {
+      operation_visitor o_v;
+      oho->op.visit(o_v);
+
+      vs.fee_data.asset = o_v.fee_asset;
+      vs.fee_data.amount = o_v.fee_amount;
+      vs.transfer_data.asset = o_v.transfer_asset_id;
+      vs.transfer_data.amount = o_v.transfer_amount;
+      vs.transfer_data.from = o_v.transfer_from;
+      vs.transfer_data.to = o_v.transfer_to;
+   }
+
+   // block data
+   std::string trx_id = "";
+   if(!b.transactions.empty() && oho->trx_in_block < b.transactions.size()) {
+      trx_id = b.transactions[oho->trx_in_block].id().str();
+   }
+   block_struct bs;
+   bs.block_num = b.block_num();
+   bs.block_time = b.timestamp;
+   bs.trx_id = trx_id;
+
+   // check if we are in replay or in sync and change number of bulk documents accordingly
+   uint32_t limit_documents = 0;
+   if((fc::time_point::now() - b.timestamp) < fc::seconds(30))
+      limit_documents = _elasticsearch_bulk_sync;
+   else
+      limit_documents = _elasticsearch_bulk_replay;
+
+   if(bulk.size() < limit_documents) { // we have everything, creating bulk array
```

Comment:
This check should be removed, because in some cases it will cause data loss. For example, in last iteration, it's in replay mode, so limit is big, so bulk has a big size; then in this iteration, it switched to sync mode, so limit suddenly became small and this check will return false, so the next line won't be executed.

Reply:
you are right, removed here https://github.com/bitshares/bitshares-core/pull/444/commits/a854409dba72a4edc4719afb121a904f310e4b11